### PR TITLE
Add Terraform provider docs content

### PR DESCRIPTION
**Pull request by nrichers**
## Description

This PR adds content for the Terraform provider docs on elastic.co.

## Related Issues

Relates to 
* https://github.com/elastic/terraform-provider-ec/issues/145
* https://github.com/elastic/docs/pull/1995

## How Has This Been Tested?

Built locally, looks like this: 

image

~(Note the double horizontal lines. Need to look into what we need to get rid of one of these.)~ Fixed this with Greg's help.

## Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (improves code quality but has no user-facing effect)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation

## Readiness Checklist

- [ ] My code follows the code style of this project
- [ ] My change requires a change to the documentation
- [x] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed

### CODE REVIEW

**Review by karencfv — COMMENTED** [review]
Thanks for the PR @nrichers! I have only one concern. As the `docs/` directory is what is visible on the terraform registry, I am wondering the implications of having a second file named `index` and a file with the `asciidoc` extension. 

As far as I know we can't ask anyone from the Hashicorp side about this until the legal stuff for the partnership is done. Is it possible for this file to live in a new `docs-internal/` directory or something similar?
I am not entirely sure

File: docs/index.asciidoc [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+:n: Elastic Cloud Terraform provider
+:p: tpec
+
+include::{asciidoc-dir}/../../shared/attributes.asciidoc[]
+
+The {n} allows you to provision Elastic Cloud deployments on any Elastic Cloud platform, whether it’s Elasticsearch Service or Elastic Cloud Enterprise.
+
+The provider lets you manage Elastic Cloud deployments as code, and introduce DevOps-driven methodologies to manage and deploy the Elastic Stack and solutions.
+
+To get started, see our link:https://registry.terraform.io/providers/elastic/ec/latest/docs[Elastic Cloud Terraform provider documentation].
```
Comment (karencfv) [inline docs/index.asciidoc:11]: This link is not live yet, would it break the docs build?

**Comment by alaudazzi** [comment]
@nrichers can we check with @gtback if the doc build accepts file types other than Asciidoc?

**Review by Kushmaro — APPROVED** [review]
text LGTM

File: docs/index.asciidoc (hunk 1 above)
Comment (nrichers) [reply docs/index.asciidoc:11]: External links are not checked by our elastic.co docs build, so this should be fine.

Comment (gtback) [reply docs/index.asciidoc]: ☝️

**Comment by gtback** [comment]
> @nrichers can we check with @gtback if the doc build accepts file types other than Asciidoc?

We currently only support Asciidoc. I suppose we could hack in support for copying raw HTML files (we have some minimal support for overriding the "index.html" page for a book right now), but I imagine that gets us further from our goal than closer (maintaining Markdown and HTML). Perhaps we could try automating Pandoc conversion of Markdown -> Asciidoc, as some other projects do? Alternatively, if Hashicorp will host what becomes the "primary" version of the docs, perhaps we should just have a small overview and a link to Hashicorp's site, like we do with some of the Elasticsearch client library pages:
* [Go](https://www.elastic.co/guide/en/elasticsearch/client/go-api/current/overview.html)
* [Ruby](https://www.elastic.co/guide/en/elasticsearch/client/ruby-api/current/ruby_client.html)
* [Rust](https://www.elastic.co/guide/en/elasticsearch/client/rust-api/current/overview.html)

**Comment by nrichers** [comment]
> As far as I know we can't ask anyone from the Hashicorp side about this until the legal stuff for the partnership is done. Is it possible for this file to live in a new `docs-internal/` directory or something similar?

@karencfv moving the files is not a problem, I've updated the PRs. I moved the content to `docs-elastic/`, as the internal bit might throw off some people. Happy to move again if need be.

**Comment by nrichers** [comment]
> > @nrichers can we check with @gtback if the doc build accepts file types other than Asciidoc?
> 
> We currently only support Asciidoc. I suppose we could hack in support for copying raw HTML files (we have some minimal support for overriding the "index.html" page for a book right now), but I imagine that gets us further from our goal than closer (maintaining Markdown and HTML). Perhaps we could try automating Pandoc conversion of Markdown -> Asciidoc, as some other projects do?

We do automatically convert Markdown to Asciidoc for ecctl command reference content, but it would require additional steps to adapt this as the primary content source here. It may not be worth the effort, given that we have a short timeline and we'll be moving to a new docs system soon.

(Reading between the lines, I do wonder if @alaudazzi wasn't thinking about our forthcoming use of MDX or about using Markdown as an early experiment. She's out this week, but perhaps could comment further when she is back.)

> Alternatively, if Hashicorp will host what becomes the "primary" version of the docs, perhaps we should just have a small overview and a link to Hashicorp's site, like we do with some of the Elasticsearch client library pages

@gtback that's exactly what we are doing with this PR. I think that's likely the best approach for now, given that we use it elsewhere (I borrowed from the Site Search docs, which do something very similar.)

**Comment by gtback** [comment]
> @gtback that's exactly what we are doing with this PR. I think that's likely the best approach for now, given that we use it elsewhere (I borrowed from the Site Search docs, which do something very similar.)

Perfect! I wasn't sure if this was just the starting point and if you were planning to expand to the rest of the Markdown files in `/docs` (or if that's what @alaudazzi was asking about).

**Review by karencfv — APPROVED** [review]
Thanks everyone! @nrichers the `docs-elastic` directory is fine with me :)

**Comment by alaudazzi** [comment]
Agreed. Thank you @gtback for your feedback. As @nrichers stated in his last comment above, this is the best approach for now.

**Review by alaudazzi — APPROVED** [review]
LGTM.

**Comment by nrichers** [comment]
> LGTM.

Why are you approving PRs on your week off, @alaudazzi?! 😁

**Comment by alaudazzi** [comment]
@nrichers it was a special week off.

**Review by marclop — APPROVED** [review]

**Comment by nrichers** [comment]
Please don't merge this, yet. There are some additional horizontal rules that we need to get rid of. It looks like other books did this by way of injecting some HTML, but I'm still looking into what's required.

**Comment by gtback** [comment]
I marked this as a draft so it doesn't accidentally get merged prematurely.

**Comment by marclop** [comment]
@nrichers we're going forward with the `v0.1.0-beta` version name, I've commented in elastic/docs#1995 to update it with the `v0.1.0-beta` branch name.

**Comment by nrichers** [comment]
OK, @gtback helped me solve the issue of the extra horizontal lines (fixed via an update to https://github.com/elastic/docs/pull/1995). This PR is good to merge.

**Comment by nrichers** [comment]
> @nrichers we're going forward with the `v0.1.0-beta` version name, I've commented in [elastic/docs#1995](https://github.com/elastic/docs/pull/1995) to update it with the `v0.1.0-beta` branch name.

@marclop I've created the `v0.1.0-beta` branch off master and will make sure we can build this as part of [elastic/docs#1995].

**Pull request merged**